cobbler/modules/managers/isc.py

```diff
@@ -225,36 +215,215 @@ def write_dhcp_file(self):
+                # architecture cannot be differed due to missing bits...
+                if distro is not None and not interface.get("filename"):
+                    if distro.arch == "ppc":
+                        interface["filename"] = "grub/grub.ppc"
+                    elif distro.arch == "ppc64":
+                        interface["filename"] = "grub/grub.ppc64"
+                    elif distro.arch == "ppc64le":
+                        interface["filename"] = "grub/grub.ppc64le"
+                    elif distro.arch == "aarch64":
+                        interface["filename"] = "grub/grubaa64.efi"
+
+                if not self.settings.always_write_dhcp_entries:
+                    if not interface["netboot_enabled"] and interface['static']:
+                        continue
+
+                if dhcp_tag == "":
+                    dhcp_tag = blended_system.get("dhcp_tag", "")
+                    if dhcp_tag == "":
+                        dhcp_tag = "default"
+
+                if dhcp_tag not in dhcp_tags:
+                    dhcp_tags[dhcp_tag] = {
+                        mac: interface
+                    }
+                else:
+                    dhcp_tags[dhcp_tag][mac] = interface
+
+        # Remove macs from redundant slave interfaces from dhcp_tags otherwise you get duplicate ip's in the installer.
+        for dt in list(dhcp_tags.keys()):
+            for m in list(dhcp_tags[dt].keys()):
+                if m in ignore_macs:
+                    del dhcp_tags[dt][m]
+
+        # we are now done with the looping through each interface of each system
+        metadata = {
+            "date": time.asctime(time.gmtime()),
+            "next_server_v6": self.settings.next_server,
+            "dhcp_tags": dhcp_tags
+        }
+
+        if self.logger is not None:
+            self.logger.info("generating %s" % self.settings_file_v6)
+        self.templar.render(template_data, metadata, self.settings_file_v6, None)
+
+
+    def restart_dhcp(self, service_name):
         """
-        This syncs the dhcp server with it's new config files. Basically this restarts the service to apply the changes.
+        This syncs the dhcp server with it's new config files.
+        Basically this restarts the service to apply the changes.
         """
-        service_name = utils.dhcp_service_name()
-        if self.settings.restart_dhcp:
-            rc = utils.subprocess_call(self.logger, "dhcpd -t -q", shell=True)
-            if rc != 0:
-                error_msg = "dhcpd -t failed"
-                self.logger.error(error_msg)
-                raise CX(error_msg)
-            service_restart = "service %s restart" % service_name
-            rc = utils.subprocess_call(self.logger, service_restart, shell=True)
-            if rc != 0:
-                error_msg = "%s failed" % service_name
-                self.logger.error(error_msg)
-                raise CX(error_msg)
+        rc = 0
```

Comment:
Please make it more meaningful. If you take it over, don't forget to change the rest of the vars.
```suggestion
        return_code_service_restart = 0
```